pkg/component/component.go

```diff
@@ -82,7 +83,8 @@ type MasterConfig struct {
 
 // NodeConfig is used to backup/restore/configure worker nodes (backup and restore have an empty implementation right now)
 type NodeConfig struct {
-	CloudProvider string `mapstructure:"caKeyFilename"`
+	CloudProvider string        `mapstructure:"caKeyFilename"`
+	RetryMaxMin   time.Duration `mapstructure:""joinCmdMaxRetryMin`
```

Comment:
Can we find a better name for the config option?

In addition, do you think it could make sense to provide a reasonable default?

Reply:
I know the name of variable is one of the hardest thing to do with the cache invalidation. Please if you have a better one, please feel free to suggest! About the default value, usually I am against defaults. But if you all are agree on a default, we can set one!

Reply:
Could `joinTimeout` be a good name?

About the default, as I think we'll always end up using the same value everywhere, why not choose a reasonable default?

What do you think?

cc @phisco @ralgozino @angelbarrera92

Reply:
30min could be something acceptable? Or more?

Reply:
btw @lnovara  `joinTimeout` is surely a better name :D

Reply:
Yes, 30 minutes should be an acceptable default from my POV.

Reply:
I agree. 30 minutes is a sane default and `joinTimeout` is much better :)

Reply:
perfect. i just made both the modifications:
1. renaming the awful var type in the struct
2. adding the default logic for joinTimeout

Let me know. @angelbarrera92  when you want, we can test, if you have time, otherwise let's schedule a 30min timeframe to do that togheter 👍